src/System.Private.ServiceModel/tools/test/Bridge/Bridge/Program.cs

```diff
@@ -253,56 +284,90 @@ private static void ResetBridge(CommandLineArguments commandLineArgs)
             BridgeController.ReleaseAllResources(force: false);
         }
 
-        // Checks whether the Bridge is running and starts it if necessary
-        private static void RequireBridge(CommandLineArguments commandLineArgs)
+        // Checks whether the Bridge is running and starts it if necessary.
+        // Returns the exit code to return to the calling script
+        private static int RequireBridge(CommandLineArguments commandLineArgs)
         {
             string errorMessage = null;
+            Dictionary<string, string> properties;
 
-            if (PingBridge(commandLineArgs.BridgeConfiguration.BridgeHost,
-                                           commandLineArgs.BridgeConfiguration.BridgePort,
-                                           out errorMessage))
+            // In a test situation, the Bridge may have been launched and is still in the
+            // process of starting.  A successful ping means it is healthy, but we will retry
+            // if the ping fails and we detect the Bridge process is running.
+            DateTime startTime = DateTime.Now;
+            while (((DateTime.Now - startTime).TotalSeconds) < commandLineArgs.RequireBridgeTimeoutSeconds)
             {
-                Console.WriteLine("The Bridge is already running.");
-                Environment.Exit(0);
+                // Ping first in case the Bridge is running already or on another machine
+                if (PingBridge(commandLineArgs.BridgeConfiguration.BridgeHost,
+                                               commandLineArgs.BridgeConfiguration.BridgePort,
+                                               out errorMessage,
+                                               out properties))
+                {
+                    BridgeConfiguration bridgeConfiguation = new BridgeConfiguration(properties);
+                    Console.WriteLine("The Bridge is already running.");
+                    return Bridge_ExitCode_AlreadyRunning;
+                }
+
+                Console.WriteLine("  -- the ping failure response was: {0}", errorMessage);
+
+                // Ping failed.  If the Bridge process is running, try again until it responds.
+                // If the Bridge process is not even running, no need to keep retrying.
+                // We have to consider the Bridge.exe running that is executing this code
+                // as well as the Bridge.exe running as a service hoster.
+                bool bridgeRunning = Process.GetProcessesByName("Bridge").Count() >= 2;
```

Comment:
Just a thought - are there any other known programs that might be running on a dev machine also called "Bridge"? It's quite a generic name.